### PR TITLE
RDS Universal Tagging

### PR DESCRIPTION
**Pull request by salwandhruvdev**
This is related to #1544 
- Use Universal Tagging for RDS
- Refactor test cases

### CODE REVIEW

**Comment by salwandhruvdev** [comment]
Fixing few test cases.

**Comment by salwandhruvdev** [comment]
Ready for review. Thanks

File: tests/test_rds.py [hunk 1]
```diff
@@ -983,7 +983,7 @@ def get_dbid(self, recording, flight_data):
                 'placebo', flight_data, 'rds.CreateDBInstance_1.json')
             pill = json.load(open(pill_path))
             return pill['data']['DBInstance']['DBInstanceIdentifier']
-
+    #
```
Comment (alfredgamulo) [inline tests/test_rds.py]: stray #

File: c7n/resources/rds.py [hunk 2]
```diff
@@ -118,6 +118,7 @@ class resource_type(object):
     _generate_arn = None
     retry = staticmethod(get_retry(('Throttled',)))
     permissions = ('rds:ListTagsForResource',)
```
Comment (alfredgamulo) [inline c7n/resources/rds.py]: This can probably be removed now.

File: c7n/resources/rds.py [hunk 3]
```diff
@@ -78,8 +80,6 @@
 filters = FilterRegistry('rds.filters')
 actions = ActionRegistry('rds.actions')
 
-filters.register('tag-count', tags.TagCountFilter)
-filters.register('marked-for-op', tags.TagActionFilter)
 filters.register('health-event', HealthEventFilter)
 actions.register('auto-tag-user', AutoTagUser)
```
Comment (ewbankkit) [inline c7n/resources/rds.py]: I think that this can be removed as it's done in `register_universal_tags`.

File: c7n/resources/rds.py [hunk 4]
```diff
@@ -117,7 +117,7 @@ class resource_type(object):
     action_registry = actions
     _generate_arn = None
     retry = staticmethod(get_retry(('Throttled',)))
-    permissions = ('rds:ListTagsForResource',)
+    augment = universal_augment
```
Comment (ewbankkit) [inline c7n/resources/rds.py]: Any reason why the `rds-snapshot` resource wasn't changed?
It supports universal tagging also:
* http://docs.aws.amazon.com/awsconsolehelpdocs/latest/gsg/supported-resources.html

I suggest opening a second PR if you decide to do that.

Comment (kapilt) [reply c7n/resources/rds.py]: yeah definitely a second pr for that, the volume of line change on these with the test data surgery is large.

File: tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_1.json [hunk 5]
```diff
@@ -1,52 +1,52 @@
 {
-    "status_code": 200, 
+    "status_code": 200,
```
Comment (kapilt) [inline tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_1.json]: this file shouldn't be needed anymore. does the test pass if you remove?

File: tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_2.json [hunk 6]
```diff
@@ -1,52 +1,52 @@
 {
-    "status_code": 200, 
+    "status_code": 200,
```
Comment (kapilt) [inline tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_2.json]: all of the ListTagsForResource in the test directory should be obsolete with this pr.

File: tests/data/placebo/test_rds_tag_and_remove/rds.ListTagsForResource_1.json [hunk 7]
```diff
@@ -2,21 +2,32 @@
     "status_code": 200, 
     "data": {
         "ResponseMetadata": {
+            "RetryAttempts": 0,
```
Comment (kapilt) [inline tests/data/placebo/test_rds_tag_and_remove/rds.ListTagsForResource_1.json]: this one should be gone too?

File: tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_1.json (hunk 5 above)
Comment (salwandhruvdev) [reply tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_1.json]: No Kapil the test fails if I remove this file.  We are querying `rds-snapshot` resource in this test which is still using `list_tags_for_resource`:
```
snaps = p.resource_manager.get_resource_manager(
            'rds-snapshot').get_resources(('final-mxtest-2017-05-25',))
```

Comment (salwandhruvdev) [reply tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_1.json]: so we would need one response for listing the tags for this test. I will delete the other one.

File: tests/data/placebo/test_rds_tag_and_remove/rds.ListTagsForResource_1.json (hunk 7 above)
Comment (salwandhruvdev) [reply tests/data/placebo/test_rds_tag_and_remove/rds.ListTagsForResource_1.json]: This test contains a separate call for listing the tags. I will replace that with GetResources.

Comment (kapilt) [reply tests/data/placebo/test_rds_tag_and_remove/rds.ListTagsForResource_1.json]: if it was the test code doing it explicitly via client in the test case thats fine to leave as is

File: tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_1.json (hunk 5 above)
Comment (kapilt) [reply tests/data/placebo/test_rds_delete_copy_restore/rds.ListTagsForResource_1.json]: that's fine then if its a correlated resource. note that will likely mean re-recording this when doing snapshots.

**Comment by JohnTheodore** [comment]
@salwandhruvdev fyi https://gitter.im/capitalone/cloud-custodian

File: c7n/resources/rds.py [hunk 8]
```diff
@@ -130,58 +129,10 @@ def generate_arn(self):
...
-            self.manager.get_model(),
-            dbs,
-            self.manager.session_factory,
-            self.manager.executor_factory,
-            self.manager.generate_arn,
-            self.manager.retry))
-        return dbs
-
-
-class ConfigRDS(ConfigSource):
```
Comment (kapilt) [inline c7n/resources/rds.py]: umm.. this stuff shouldn't be removed, you want the universal tag augment in the DescribeRDS source, the ConfigRDS stays as is, please put back get_source, DescribeRDS, and ConfigRDS. you can yank _rds_tags.

**Pull request merged**